mspray/apps/main/tests/views/test_performance.py

```diff
@@ -1,36 +1,297 @@
+        request = factory.get("/performance/spray-operators/2939/102/summary")
+        view = SprayOperatorSummaryView.as_view()
+        response = view(
+            request,
+            team_leader=spray_operator.team_leader.id, slug=spray_operator.id)
+
+        self.assertEqual(response.status_code, 200)
+
+        """
+        Query obtains all data for the SprayOperator
+        including submissions made and passes this to the serializer
+        """
+        queryset = SprayOperator.objects.raw(
+            SOP_PERFORMANCE_SQL, [spray_operator.team_leader_assistant.id]
+        )
+        serializer = SprayOperatorPerformanceReportSerializer(
+            queryset, many=True
+        )
+        result = {
+            'other': 0,
+            'refused': 6,
+            'sprayed': 2,
+            'sprayable': 19,
+            'not_sprayable': 0,
+            'not_sprayed_total': 6,
+            'data_quality_check': False,
+            'found_difference': 15,
+            'sprayed_difference': 15,
+            'no_of_days_worked': 2,
+            'avg_structures_per_so': 9.5,
+            'avg_start_time': datetime.time(5, 27, 25, 666667),
+            'avg_end_time': datetime.time(5, 32, 42, 666667)}
+
+        self.assertEqual(response.context_data['data'], serializer.data)
+        self.assertEqual(response.context_data['totals'], result)
+
+    def test__spray_operator_daily_view(self):
+        """Test SprayOperatorDailyView.
+
+        This is done by confirming the data received in the response
+        context is being aggregated appropriately within the view
+        """
+        # first we load our test data
+        data_setup()
+        self._load_fixtures()
+        rhc = Location.objects.get(name='John')
+        spray_operator = SprayOperator.objects.first()
+        team_leader = TeamLeader.objects.first()
+        spray_operator.team_leader = team_leader
+        spray_operator.save()
+
+        # next we identify a SprayDay object for the spray operator
+        spray_day = SprayDay.objects.filter(spray_operator=spray_operator)
+        spray_day.update(sprayable=True)
+
+        factory = RequestFactory()
+        request = factory.get(
+            "/performance/spray-operators/2939/102/538/daily")
+        view = SprayOperatorDailyView.as_view()
+
+        """
+        Create performance report objects from the submissions made
+        by the particular sprayoperator.
+        """
+        performance_report(spray_operator)
+        report1 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report1.found = 7
+        report1.reported_sprayed = 16
+        report1.reported_found = 22
+        report1.save()
+
+        report2 = PerformanceReport.objects.get(spray_operator=spray_operator)
+        report2.id = None
+        report2.sprayformid = 7658
+        report2.refused = 6
+        report2.found = 12
+        report2.reported_sprayed = 6
+        report2.save()
+
+        """
```

Comment:
I would prefer comments made with `#` as opposed to a string as done here when the comments are in the middle of a block of code.  Would you agree?